File: Leetcode/392/392.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        if (s.empty()) return true;
        if (t.empty()) return false;
        auto iter = s.begin();
        for (const auto &i : t) {
            if (iter == s.end()) return true;
            if (i == *iter) {
                ++iter;
            }
        }
        return (iter == s.end());
    }
};

File: Leetcode/392/392.go
func isSubsequence(s string, t string) bool {
	j := 0
	for i := 0; i < len(s); {
		if j == len(t) {
			return false
		}
		if t[j] == s[i] {
			i++
		}
		j++
	}
	return true
}